stdlib/src/builtin/string.mojo

```diff
@@ -553,7 +553,7 @@ struct String(
 
     # TODO: Remove this method when #2317 is done
     @always_inline
-    fn __init__(inout self, owned impl: Self._buffer_type):
+    fn __init__(inout self, owned impl: List[Int8]):
```

Comment:
**Suggestion** Should we throw @deprecated on this then given the note below?

>
Note that you should use the constructor from `List[UInt8]` instead

Reply:
I agree. Maybe let's do that in another PR, I'll also remove the unused methods at the same time.